Lambdas/GHHooks/EventHandler/EventHandler.swift

```diff
@@ -0,0 +1,53 @@
+import DiscordBM
+
+struct EventHandler {
+    let context: HandlerContext
+
+    func handle() async throws {
+        switch context.eventName {
+        case .issues:
+            try await IssueHandler(context: context).handle()
+        case .pull_request:
+            try await PRHandler(context: context).handle()
+        case .ping:
+            try await onPing()
+        default:
+            try await onDefault()
+        }
+    }
+
+    func onPing() async throws {
+        try await context.discordClient.createMessage(
+            channelId: Constants.Channels.logs.id,
+            payload: .init(embeds: [.init(
+                title: "Ping events should not reach here",
+                description: """
+                Ping events must be handled immediately, even before any body-decoding happens.
+                Action: \(context.event.action ?? "null")
+                Repo: \(context.event.repository.name)
+                """,
+                color: .red
+            )])
+        ).guardSuccess()
+    }
+
+    func onDefault() async throws {
+        try await context.discordClient.createMessage(
+            channelId: Constants.Channels.logs.id,
+            payload: .init(embeds: [.init(
+                title: "Received UNHANDLED event \(context.eventName)",
+                description: """
+                Action: \(context.event.action ?? "null")
+                Repo: \(context.event.repository.name)
+                """,
+                color: .red
+            )])
+        ).guardSuccess()
+    }
+}
+
+extension String {
+    func unicodesPrefix(_ maxLength: Int) -> String {
+        String(self.unicodeScalars.prefix(maxLength))
+    }
+}
```

Comment:
This can, of course, result in truncated display strings - a Unicode scalar is not even remotely a grapheme cluster. For example, take 👩‍👩‍👧‍👧, which is composed of `U+1F469 WOMAN` `U+200D ZERO WIDTH JOINER` `U+1F469 WOMAN` `U+200D ZERO WIDTH JOINER` `U+1F467 GIRL` `U+200D ZERO WIDTH JOINER` `U+1F467 GIRL`. Or even more fun, 👍 (`U+1F44D THUMBS UP`) versus 👍🏿(`U+1F44D THUMBS UP` `U+1F3FF EMOJI MODIFIER FITZPATRICK TYPE-6`). Breaking up strings at any granularity finer than extended grapheme cluster boundaries can have nontrivial consequences. Yes, even in the context where I see this method being used. Please take the prefix of the string itself.

Reply:
I'm not going to act like I know all about graphemes ans stuff, but not using .unicodeScaalrs can easily result In bad-request from Discord as soon as there is an emoji in the String. .unicodeScalars has been working quite well while working with APIs for me. Not sure what else to do other than this.

Reply:
I guess i could maybe find a way to not let a half-string scalar get in ... i might or might not give it a try 🤔

Reply:
Doing it "truly correctly" would be a rather intricate and messy task, but there's a simple cheat that's `O(1)` for the vast majority of cases:
```swift
    func unicodesPrefix(_ maxUnicodeScalars: Int) -> String {
        assert(maxUnicodeScalars >= 0, "Can request a negative maximum.")

        // Take a prefix of the string (i.e. a sequence of extended grapheme clusters) first.
        // Most often, of the time this will already be a result that suits our needs.
        var trimmed = self.prefix(maxUnicodeScalars)
        
        // If the result still has too many unicode scalars, there're one or more grapheme
        // clusters in the string. Keep dropping extended grapheme clusters off the end (which
        // with `String` is as easy as just removing the last `Character`) until we're within
        // bounds. Worst-case complexity is `O(n)`.
        while trimmed.unicodeScalars.count > maxUnicodeScalars {
            trimmed.removeLast()
        }
        
        return String(trimmed)
    }
```

Bonus chatter:

An "extended grapheme cluster" is any sequence of Unicode codepoints (in Swift parlance, a `UnicodeScalar`) which, when read in order from the beginning and applying Unicode's rules for combining characters, variation selectors, etc., yields a single "printable" glyph (where "printable" is a rather flexible term including such visually questionable esoterica as non-breaking spaces, LTR/RTL selectors, and so forth). Swift calls these `Character`s.

A "grapheme cluster" (formally, a "legacy grapheme cluster") is the same thing, except using the legacy (Unicode 3.2 or so) combining rules. These exist now only as a historical curiosity, not unlike the original FSS-UTF 6-byte version of UTF-8; you'll never have to worry about the difference.